api/tests/test_search_api.py

```diff
@@ -0,0 +1,310 @@
+# coding: utf-8
+import pytest
+from fastapi.testclient import TestClient
+
+from feeds_gen.models.search_feeds200_response import SearchFeeds200Response  # noqa: F401
+from .test_utils.database import TEST_GTFS_FEED_STABLE_IDS, TEST_GTFS_RT_FEED_STABLE_ID
+
+
+@pytest.mark.parametrize(
+    "search_query",
+    [
+        # Provider
+        "MobilityDataTest provider",
+        # Provider using a lexically similar word
+        "MobilityDataTest PROVIDE",
+        # Feed name
+        "MobilityDataTest Feed name",
+        # Feed name using a lexically similar word
+        "MobilityDataTest feedING",
+    ],
+)
+def test_search_feeds_all_feeds(client: TestClient, search_query: str):
+    """
+    Retrieve all feeds with a search query using provider or feed name.
+    """
+    params = [("limit", 100), ("offset", 0), ("feed_id", ""), ("data_type", ""), ("search_query", search_query)]
+    headers = {
+        "Authentication": "special-key",
+    }
+    response = client.request(
+        "GET",
+        "/v1/search",
+        headers=headers,
+        params=params,
+    )
+
+    # Assert the status code of the HTTP response
+    assert response.status_code == 200
+
+    # Parse the response body into a Python object
+    response_body = SearchFeeds200Response.parse_obj(response.json())
+
+    assert response_body.total == len(TEST_GTFS_FEED_STABLE_IDS)
+    assert len(response_body.results) == len(TEST_GTFS_FEED_STABLE_IDS)
```

Comment:
what is the reasoning behind returning the property `total` if it can be acquired through `results.length`

Reply:
The _total_ property contains the total of the _query_ without considering _limit_ and _offset_(pagination). The total is the same in this specific test with or without pagination.